File: Lab13/Lab13.4/app/src/main/java/com/example/customgridview/MyArrayAdapter.java
package com.example.customgridview;

import android.app.Activity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;
import java.util.ArrayList;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class MyArrayAdapter extends ArrayAdapter<Image> {
    private Activity context;
    private int Layoutid;
    private ArrayList<Image> myArray;

    // Constructor này dùng để khởi tạo các giá trị
    // @param context: là Activity truyền vào
    // @param layoutId: là layout custom do ta tao (listitem.xml)
    // @param arr: Danh sách đối tượng truyền từ Main
    public MyArrayAdapter(Activity context, int Layoutid, ArrayList<Image> arr) {
        super(context, Layoutid, arr);
        this.context = context;
        this.Layoutid = Layoutid;
        this.myArray = arr;
    }

    // Gọi đến hàm getView để xây dựng lại Adapter
    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        LayoutInflater inflater = context.getLayoutInflater();
        convertView = inflater.inflate(Layoutid, null);

        ImageView imgItem = convertView.findViewById(R.id.imageView1); // Ánh xạ ImageView
        TextView txtName = convertView.findViewById(R.id.textView1); // Ánh xạ TextView

        Image myImage = myArray.get(position); // Lấy đối tượng Image
        imgItem.setImageResource(myImage.getImg()); // Thiết lập hình ảnh
        txtName.setText(myImage.getName()); // Thiết lập tên ảnh

        return convertView;
    }
}

File: Lab24/app/src/main/java/com/example/dongabank/MyArrayAdapter.kt
package com.example.dongabank

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView

class MyArrayAdapter(
    private val context: Activity,
    private val resource: Int,
    private val objects: List<Tygia>
) : ArrayAdapter<Tygia>(context, resource, objects) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val inflater = context.layoutInflater
        val item = inflater.inflate(resource, null)

        val tygia = objects[position]

        val imgHinh = item.findViewById<ImageView>(R.id.imgHinh)
        val txtType = item.findViewById<TextView>(R.id.txtType)
        val txtMuaTM = item.findViewById<TextView>(R.id.txtMuaTM)
        val txtBanTM = item.findViewById<TextView>(R.id.txtBanTM)
        val txtMuaCK = item.findViewById<TextView>(R.id.txtMuaCK)
        val txtBanCK = item.findViewById<TextView>(R.id.txtBanCK)

        imgHinh.setImageBitmap(tygia.bitmap)
        txtType.text = tygia.type
        txtMuaTM.text = tygia.muatienmat
        txtBanTM.text = tygia.bantienmat
        txtMuaCK.text = tygia.muack
        txtBanCK.text = tygia.banck

        return item
    }
}